File: src/main/java/io/github/saneea/dvh/feature/file/ReadFile.java
package io.github.saneea.dvh.feature.file;

import java.io.BufferedInputStream;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import io.github.saneea.dvh.Feature;

public class ReadFile//
		extends FileFeature//
		implements Feature.Out.Bin.Stream {

	private OutputStream out;

	@Override
	protected String getDescription() {
		return "transfer bytes from file to standard output";
	}

	@Override
	protected void handleFile(String fileName) throws IOException {
		try (InputStream in = new BufferedInputStream(//
				new FileInputStream(fileName))) {
			in.transferTo(out);
		}
	}

	@Override
	public void setOut(OutputStream out) {
		this.out = out;
	}

}

File: src/main/java/io/github/saneea/dvh/feature/file/ReadFile.kt
package io.github.saneea.dvh.feature.file

import io.github.saneea.dvh.Feature
import io.github.saneea.dvh.FeatureContext
import io.github.saneea.dvh.utils.transferFrom
import java.io.File
import java.io.OutputStream

class ReadFile :
    FileFeature(),
    Feature.Out.Bin.Stream {

    override lateinit var context: FeatureContext
    override lateinit var outBinStream: OutputStream

    override val description = "transfer bytes from file to standard output"

    override fun handleFile(file: File) {
        file
            .inputStream().buffered()
            .use(outBinStream::transferFrom)
    }
}

